src/Pact.hs

```diff
@@ -32,17 +33,31 @@ hasNullValue :: AT.Pair -> Bool
 hasNullValue (_, Null) = True
 hasNullValue _ = False
 
+newtype Headers = Headers [(String, String)]
```

Comment:
Type safety: If `Headers` was an alias, the type checker would silently unify it with `[(String, String)]`, regardless if that was intended or not. By introducing a newtype, we need to express our intend to convert explicitly.

The other advantage of newtypes is that we can define typeclass instances for them, which you couldn't do for `Monoid [(String, String)]`, since that would be an overlapping orphan instance.